hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/common/ChunkBuffer.java

```diff
@@ -143,6 +143,12 @@ default ChunkBuffer put(ByteString b) {
    */
   long writeTo(GatheringByteChannel channel) throws IOException;
 
+  /**
+   * Write all the contents of the buffer from the current position to the limit
+   * to {@code channel}.
+   */
+  void writeFully(GatheringByteChannel channel) throws IOException;
```

Comment:
I would like to suggest fixing `writeTo` instead of adding this new method to the interface.

1. In what case may we want to use the existing method?  With this fix, `writeTo` becomes unused in production.

2. `TestChunkBuffer.assertWrite` tests `writeTo` and expects it to write all data.  However, it only works because the channels it uses, unlike `FileChannel`, write fully.  This can be verified by randomly adjusting `limit` for the buffer in `MockGatheringChannel.write(ByteBuffer)`, i.e. simulating incomplete writes.  On the other hand, `writeFully` is not covered by unit tests.

3. We can avoid the changes in `ChunkUtils`, most of which is due to `writeFully` not returning the number of bytes written.

Reply:
@duongkame you can pick https://github.com/adoroszlai/ozone/commit/daee756ee4d09334ad9b72a5fdf4926027ff2188, which implements the suggested change